tests/test_device.py

```diff
@@ -0,0 +1,156 @@
+# MIT License
+
+BLE_DEVICE_DEFAULTS = {
+    "name": None,
+    "rssi": -127,
+    "details": None,
+}
+
+
+def generate_ble_device(
+    address: str | None = None,
+    name: str | None = None,
+    details: Any | None = None,
+    rssi: int | None = None,
+    **kwargs: Any,
+) -> BLEDevice:
+    """Generate a BLEDevice with defaults."""
+    new = kwargs.copy()
+    if address is not None:
+        new["address"] = address
+    if name is not None:
+        new["name"] = name
+    if details is not None:
+        new["details"] = details
+    if rssi is not None:
+        new["rssi"] = rssi
+    for key, value in BLE_DEVICE_DEFAULTS.items():
+        new.setdefault(key, value)
+    return BLEDevice(**new)
+
+
+# end
+
+
+class PackTestData:
+    def __init__(
+        self: PackTestData, name: str, dt: datetime, ampm: bool, packed: bytes
+    ):
+        self.name = name
+        self.dt = dt
+        self.ampm = ampm
+        self.packed = packed
+
+    def test(self: PackTestData) -> tuple[bool, bytes, bytes, str]:
+        result = ThermoProDevice.pack_datetime(self.dt, self.ampm)
+        return result == self.packed, result, self.packed, self.name
+
+    def generate(self: PackTestData) -> None:
+        print(
+            f"""PackTestDate("{self.name}", datetime.fromisoformat"""
+            f"""("{self.dt.replace(microsecond=0).isoformat()}"), {self.ampm}, """
+            f"""{ThermoProDevice.pack_datetime(self.dt, self.ampm)!r}),"""
+        )
+
+
+now = datetime.now()
+
+PACK_TESTDATA = [
+    PackTestData(
+        "self-test-24hour", now, False, ThermoProDevice.pack_datetime(now, False)
+    ),
+    PackTestData(
+        "self-test-12hour", now, True, ThermoProDevice.pack_datetime(now, True)
+    ),
+    PackTestData(
+        "2025-01-24-20:54:15-12",
+        datetime.fromisoformat("2025-01-24T20:54:15"),
+        True,
+        b"\xa5\x19\x01\x18\x146\x0f\x05\x00Z",
+    ),
+    PackTestData(
+        "2025-01-24-20:54:15-24",
+        datetime.fromisoformat("2025-01-24T20:54:15"),
+        False,
+        b"\xa5\x19\x01\x18\x146\x0f\x05\x01Z",
+    ),
+]
+
+
+class ThermoProDeviceMock(ThermoProDevice):
+    async def connect(self: ThermoProDeviceMock) -> BleakClient:
+        client = BleakClient("")
+        client.write_gatt_char = AsyncMock()
+        client.disconnect = AsyncMock()
+        self.client = client
+        return client
+
+
+class ThermoProDeviceTest(IsolatedAsyncioTestCase):
+    def test_can_create(self: ThermoProDeviceTest) -> None:
+        ThermoProDevice(generate_ble_device("aa:bb:cc:dd:ee:ff", "TP358"))
+
+    def test_pack_cases(self: ThermoProDeviceTest) -> None:
+        # base code for this section comes from - https://stackoverflow.com/a/52884914
+        results = Counter()  # type: Counter[bool]
+        for idx, pack_test in enumerate(PACK_TESTDATA):
+            with self.subTest(i=idx):
+                passed, result, packed, name = pack_test.test()
+                self.assertTrue(
```

Comment:
Please use bare asserts. Sorry https://docs.astral.sh/ruff/rules/pytest-unittest-assertion/#pytest-unittest-assertion-pt009 isn't turned on or it would have already shown this

https://docs.astral.sh/ruff/rules/pytest-unittest-assertion/